target_worker/__init__.py

```diff
@@ -0,0 +1,50 @@
+"""Target Worker library.
+
+With the help of this library a target worker can be assigned on-the-fly
+The library also provides all necessary information to the worker to process
+the GET requests
+"""
+
+import logging
+import os
+import sys
+
+logger = logging.getLogger()
+
+if os.environ.get('INPUT_DATA_FORMAT') == 'TOPOLOGY':
+    logger.info('Target Worker is Topology')
+
+    USERNAME = os.environ.get('USERNAME')
+    PASSWORD = os.environ.get('PASSWORD')
+    ENDPOINT = os.environ.get('TOPOLOGY_INVENTORY_ENDPOINT')
+
+    if None in [USERNAME, PASSWORD, ENDPOINT]:
+        logger.error('Environment not set properly, '
+                     'missing USERNAME or PASSWORD or '
+                     'TOPOLOGY_INVENTORY_ENDPOINT')
+        sys.exit(1)
+
+    NAME = 'worker_topology'
+    INFO = {
+        'authentication': {
+            'username': USERNAME,
+            'password': PASSWORD,
+            'endpoint': ENDPOINT
```

Comment:
Factual note: Endpoint is not an authentication.

And if you list that separately, you can than have the `INFO` dict in this format:
```python
INFO = {
    'auth': (USERNAME, PASSWORD),
    'endpoint': ENDPOINT,
    ...
}
```

Ant that can help you later in the code, because you don't need to unpack the values at [`workers.py` L130](https://github.com/ManageIQ/aiops-data-collector/pull/20/files#diff-738ac49b5cddf3f7ce9f1a17ec48f249R130) and use it later at the [`_retryable()` call](https://github.com/ManageIQ/aiops-data-collector/pull/20/files#diff-738ac49b5cddf3f7ce9f1a17ec48f249R130) as simple as:
```python
_retryable(
    'get',
    f'{topology_info["endpoint"]}/{entity}'  # or '{}/{}'.format(topology_info["endpoint"], entity), which might be more readable
    auth=topology_info["auth"],
    ...
)
```

What do you think? I think it's nicer, easier to follow and shorter as well. :wink:

Reply:
Yep, good idea to separate out `ENDPOINT` and also a tuple for `auth` which can be later used as is in workers.py

Although as a side note I would like to add that many times, a few extra variables (and sometimes unnecessary) are created just to keep the lines shorter and pylama happy :)